Source/Core/Core/PowerPC/JitArm64/JitArm64_Integer.cpp

```diff
@@ -2004,9 +2009,11 @@ void JitArm64::srawx(UGeckoInstruction inst)
   }
   else
   {
-    gpr.BindToRegister(a, a == b || a == s);
+    bool allocate_reg = a == b || a == s;
+    gpr.BindToRegister(a, allocate_reg);
 
-    ARM64Reg WA = gpr.GetReg();
+    allocate_reg = allocate_reg || js.op->wantsCA;
```

Comment:
The code would be clearer if you declared a new variable instead of modifying `allocate_reg` here. (The name `allocate_reg` makes perfect sense for this value, but not much sense for the value that's being passed to `BindToRegister`.)